tensorflow/core/kernels/data/batch_dataset_op_test_v2.cc

```diff
@@ -0,0 +1,640 @@
+    TF_RETURN_IF_ERROR(DatasetOpsTestBase::CheckDatasetCardinality(
+        *batch_dataset_, expected_cardinality));
+    return Status::OK();
+  }
+
+  Status CheckDatasetSave() {
+    TF_RETURN_IF_ERROR(DatasetOpsTestBase::CheckDatasetSave(*batch_dataset_));
+    return Status::OK();
+  }
+
+  Status CheckDatasetIsStateful(bool expected_stateful) {
+    TF_RETURN_IF_ERROR(DatasetOpsTestBase::CheckDatasetIsStateful(
+        *batch_dataset_, expected_stateful));
+    return Status::OK();
+  }
+
+  Status CheckIteratorOutputDtypes(
+      const DataTypeVector& expected_output_dtype) {
+    TF_RETURN_IF_ERROR(DatasetOpsTestBase::CheckIteratorOutputDtypes(
+        *iterator_, expected_output_dtype));
+    return Status::OK();
+  }
+
+  Status CheckIteratorOutputShapes(
+      const std::vector<PartialTensorShape>& expected_output_shapes) {
+    TF_RETURN_IF_ERROR(DatasetOpsTestBase::CheckIteratorOutputShapes(
+        *iterator_, expected_output_shapes));
+    return Status::OK();
+  }
+
+  Status CheckIteratorPrefix(const string& expected_iterator_prefix) {
+    TF_RETURN_IF_ERROR(DatasetOpsTestBase::CheckIteratorPrefix(
+        *iterator_, expected_iterator_prefix));
+    return Status::OK();
+  }
+
+  Status CheckIteratorSaveAndRestore(
+      const string& iterator_prefix,
+      const std::vector<Tensor>& expected_outputs,
+      const std::vector<int>& breakpoints) {
+    TF_RETURN_IF_ERROR(DatasetOpsTestBase::CheckIteratorSaveAndRestore(
+        *batch_dataset_, iterator_ctx_.get(), iterator_prefix, expected_outputs,
+        breakpoints));
+    return Status::OK();
+  }
+
+ protected:
+  // Creates a new `BatchDataset` op kernel.
+  Status CreateBatchDatasetOpKernel(
+      const BatchDatasetParams& dataset_params,
+      std::unique_ptr<OpKernel>* batch_dataset_op_kernel) {
+    name_utils::OpNameParams params;
+    params.op_version = kOpVersion;
+    NodeDef node_def = test::function::NDef(
+        kNodeName, name_utils::OpName(BatchDatasetOp::kDatasetType, params),
+        {BatchDatasetOp::kInputDataset, BatchDatasetOp::kBatchSize,
+         BatchDatasetOp::kDropRemainder},
+        {{BatchDatasetOp::kParallelCopy, dataset_params.parallel_copy},
+         {BatchDatasetOp::kOutputTypes, dataset_params.output_dtypes},
+         {BatchDatasetOp::kOutputShapes, dataset_params.output_shapes}});
+    TF_RETURN_IF_ERROR(CreateOpKernel(node_def, batch_dataset_op_kernel));
+    return Status::OK();
+  }
+
+  // Create a new `BatchDataset` op kernel context
+  Status CreateBatchDatasetContext(
+      OpKernel* const op_kernel,
+      gtl::InlinedVector<TensorValue, 4>* const inputs,
+      std::unique_ptr<OpKernelContext>* context) {
+    TF_RETURN_IF_ERROR(CheckOpKernelInput(*op_kernel, *inputs));
+    TF_RETURN_IF_ERROR(CreateOpKernelContext(op_kernel, inputs, context));
+    return Status::OK();
+  }
+
+ private:
+  BatchDatasetParams batch_dataset_params_;
+  std::unique_ptr<OpKernel> batch_dataset_kernel_;
```

Comment:
These member variables can be moved to the test base as every dataset op test will have them.